pkgs/top-level/all-packages.nix

```diff
@@ -7530,10 +7530,16 @@ in
 
   philter = callPackage ../tools/networking/philter { };
 
+  phoc = callPackage ../applications/misc/phoc {
+    wlroots = wlroots_0_12;
```

Comment:
I think taking `wlroots_0_12` as argument is better since that makes it obvious to anyone touching the expression that a non-standard version is used.

Reply:
I think technically its supposed to be done here so less references in phoc have to be changed in the future when phoc requires a different wlroots version. Maybe a comment could be good in phoc to explain the non-standard version.

Reply:
I think Phosh (and probably most others if they are actively maintained) will eventually switch, as 0.13 was just released less than a month ago. I say it's more of a case of wlroots breaking things rather than downstream projects requiring non-standard versions for the long-term.

Reply:
> I think technically its supposed to be done here so less references in phoc have to be changed in the future when phoc requires a different wlroots version.

True, that's an argument for keeping it the way it is.

Reply:
As I understand it, parameters to callPackage are preferred when it's a previous version that's being used not as a strict requirement, but because things won't work for the time being.

E.g. python2/python3 are arguments in the derivation's file. Not `python = python2`. So here this is the right way to deal with the problem.